raiden/transfer/architecture.py

```diff
@@ -272,6 +296,50 @@ def dispatch(self, state_change: StateChange) -> List[Event]:
 
         return events
 
+    def detect_balance_proof_change(self, old_state, current_state):
```

Comment:
this is quite hard to follow, could you please simplify this code?

Reply:
I tried to make it more explicit!